fedimint-server/src/net/peers.rs

```diff
@@ -258,9 +266,13 @@ impl<M> PeerConnectionStateMachine<M>
 where
     M: Debug + Clone,
 {
-    async fn run(mut self) {
-        while let Some(new_self) = self.state_transition().await {
-            self = new_self;
+    async fn run(mut self, task_handle: &TaskHandle) {
+        while !task_handle.is_shutting_down() {
+            if let Some(new_self) = self.state_transition().await {
```

Comment:
Would we technically need to either await a state transition or task termination here?

Reply:
Yes. Good point.

Reply:
Actually, probably not. cdf83b9e5